etna/analysis/plotters.py

```diff
@@ -1819,3 +1822,143 @@ def metric_per_segment_distribution_plot(
+    in_column:
+        column to plot
+    segments:
+        segments to use
+    columns_num:
+        number of subplots columns
+    figsize:
+        size of the figure in inches
+    start:
+        start timestamp for plot
+    end:
+        end timestamp for plot
+
+    Notes
+    -----
+    Jupyter notebook might display the results incorrectly,
+    in this case try to use ``!jupyter nbextension enable --py widgetsnbextension``.
+
+    Examples
+    --------
+    >>> from etna.datasets import TSDataset
+    >>> from etna.datasets import generate_ar_df
+    >>> from etna.analysis import plot_change_points_interactive
+    >>> from ruptures.detection import Binseg
+    >>> classic_df = generate_ar_df(periods=1000, start_time="2021-08-01", n_segments=2)
+    >>> df = TSDataset.to_dataset(classic_df)
+    >>> ts = TSDataset(df, "D")
+    >>> params_bounds = {"n_bkps": [0, 5, 1], "min_size":[1,10,3]}
+    >>> plot_change_points_interactive(ts=ts, change_point_model=Binseg, model="l2", params_bounds=params_bounds, model_params=["min_size"], predict_params=["n_bkps"], figsize=(20, 10)) # doctest: +SKIP
+    """
+    from ipywidgets import FloatSlider
+    from ipywidgets import IntSlider
+    from ipywidgets import interact
+
+    if segments is None:
+        segments = sorted(ts.segments)
+
+    cache = {}
+
+    sliders = dict()
+    style = {"description_width": "initial"}
+    for param, bounds in params_bounds.items():
+        min_, max_, step = bounds
+        if isinstance(min_, float) or isinstance(max_, float) or isinstance(step, float):
+            sliders[param] = FloatSlider(min=min_, max=max_, step=step, continuous_update=False, style=style)
+        else:
+            sliders[param] = IntSlider(min=min_, max=max_, step=step, continuous_update=False, style=style)
+
+    def update(**kwargs):
+        _, ax = prepare_axes(num_plots=len(segments), columns_num=columns_num, figsize=figsize)
+
+        key = "_".join([str(val) for val in kwargs.values()])
+
+        is_fitted = False
+
+        if key not in cache:
+            m_params = {x: kwargs[x] for x in model_params}
+            p_params = {x: kwargs[x] for x in predict_params}
+            cache[key] = {}
+        else:
+            is_fitted = True
+            bkps = cache[key]
+
+        for i, segment in enumerate(segments):
+            ax[i].cla()
+            segment_df = ts[start:end, segment, :][segment]
+            timestamp = segment_df.index.values
+            target = segment_df[in_column].values
+
+            if not is_fitted:
+                try:
+                    algo = change_point_model(model=model, **m_params).fit(signal=target)
+                    bkps = algo.predict(**p_params)
+                    cache[key][segment] = bkps
+                except BadSegmentationParameters:
+                    cache[key][segment] = None
+
+            if cache[key][segment] is not None:
+                end_time = timestamp[cache[key][segment][0] - 1]
```

Comment:
Try to avoid code duplication here, get the list of bkps and add fake bkp=0, this way you can iterate over the bkps in the loop and not draw the first segment separatly